kedro-datasets/kedro_datasets/snowflake/snowpark_dataset.py

```diff
@@ -0,0 +1,232 @@
+"""``AbstractDataSet`` implementation to access Snowflake using Snowpark dataframes
+"""
+import logging
+from copy import deepcopy
+from typing import Any, Dict
+
+import snowflake.snowpark as sp
+from kedro.io.core import AbstractDataSet, DataSetError
+
+logger = logging.getLogger(__name__)
+
+
+class SnowparkTableDataSet(AbstractDataSet):
+    """``SnowparkTableDataSet`` loads and saves Snowpark dataframes.
+
+    Example usage for the
+    `YAML API <https://kedro.readthedocs.io/en/stable/data/\
+    data_catalog.html#use-the-data-catalog-with-the-yaml-api>`_:
+
+    .. code-block:: yaml
+        weather:
+          type: kedro_datasets.snowflake.SnowparkTableDataSet
+          table_name: "weather_data"
+          database: "meteorology"
+          schema: "observations"
+          credentials: db_credentials
+          save_args:
+            mode: overwrite
+            column_order: name
+            table_type: ''
+
+    One can skip everything but "table_name" if database and
+    schema provided via credentials. Therefore catalog entries can be shorter
+    if ex. all used Snowflake tables live in same database/schema.
+    Values in dataset definition take priority over ones defined in credentials
+
+    Example:
+    Credentials file provides all connection attributes, catalog entry
+    "weather" reuse credentials parameters, "polygons" catalog entry reuse
+    all credentials parameters except providing different schema name.
+    Second example of credentials file uses externalbrowser authentication
+
+    catalog.yml
+
+    .. code-block:: yaml
+        weather:
+          type: kedro_datasets.snowflake.SnowparkTableDataSet
+          table_name: "weather_data"
+          database: "meteorology"
+          schema: "observations"
+          credentials: snowflake_client
+          save_args:
+            mode: overwrite
+            column_order: name
+            table_type: ''
+
+        polygons:
+          type: kedro_datasets.snowflake.SnowparkTableDataSet
+          table_name: "geopolygons"
+          credentials: snowflake_client
+          schema: "geodata"
+
+    credentials.yml
+
+    .. code-block:: yaml
+        snowflake_client:
+          account: 'ab12345.eu-central-1'
+          port: 443
+          warehouse: "datascience_wh"
+          database: "detailed_data"
+          schema: "observations"
+          user: "service_account_abc"
+          password: "supersecret"
+
+    credentials.yml (with externalbrowser authenticator)
+
+    .. code-block:: yaml
+        snowflake_client:
+          account: 'ab12345.eu-central-1'
+          port: 443
+          warehouse: "datascience_wh"
+          database: "detailed_data"
+          schema: "observations"
+          user: "john_doe@wdomain.com"
+          authenticator: "externalbrowser"
+
+    As of Jan-2023, the snowpark connector only works with Python 3.8
```

Comment:
I think it's worth to put this all the way at the top of the class doc string. I can imagine a lot of users would just skip reading the examples.